### PR TITLE
remove "Saved searches" link from user menu

### PR DESCRIPTION
**Pull request by sqs**
You can still get to this by going to your settings page and then clicking the "Saved searches" tab. This is a rarely used feature and does not merit inclusion in the user menu.

## Test plan

View the user menu and confirm there is no "Saved searches" item.

## App preview:

- [Web](https://sg-web-sqs-rm-saved-searches-user-menu.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.00% (-0.10 kb) | -0.00% (-0.10 kb) | 0.00% (0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/sqs/rm-saved-searches-user-menu/compare-report.html;tab=live_object) for a full comparison between the commits cc7ebfb and 7769b2a or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by umpox — APPROVED** [review]

**Comment by limitedmage** [comment]
Should we remove this feature altogether? We removed it from the homepage and now the menu, so it's very much non-discoverable.

**Comment by vovakulikov** [comment]
+1 to remove this altogether. We can pull some pings data about it, but my bet that nobody really uses this

**Comment by limitedmage** [comment]
#ask-ce thread here: https://sourcegraph.slack.com/archives/CU93UDUBV/p1677111089934889
TL;DR some bigs customers use it, we probably shouldn't remove it and instead invest into improving it and making it more discoverable.
